networking/tests/stream_reader.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use failure::Error;
+use networking::p2p::stream::{EncryptedMessageReaderBase, MessageReaderBase};
+use tezos_messages::p2p::binary_message::BinaryMessage;
+use tezos_messages::p2p::encoding::limits::BLOCK_HEADER_MAX_SIZE;
+use tezos_messages::p2p::{
+    binary_message::BinaryChunk,
+    encoding::{
+        peer::{PeerMessage, PeerMessageResponse},
+        swap::SwapMessage,
+    },
+};
+use tokio_test::io::Builder;
+
+pub mod common;
+
+use self::common::{block_header_message_encoded, data_sample, new_log, CryptoMock, PeerMock};
+
+#[async_std::test]
+#[ignore]
```

Comment:
@akoptelov 
why is this ignored?

Reply:
It will fail unless the #511 is there. I'll remove `#[ignore]` and add more tests in that PR.